SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/RegisterAutoDestroyActorsTest/RegisterAutoDestroyActorsTest.cpp

```diff
@@ -42,52 +42,48 @@ void ARegisterAutoDestroyActorsTestPart1::PrepareTest()
 		AddStep(
 			TEXT("CLIENT_ALL_CheckActorsSpawned"), FWorkerDefinition::AllClients, nullptr, nullptr,
 			[this](float DeltaTime) {
-				int NumCharactersFound = 0;
-				int NumCharactersExpected = GetNumberOfServerWorkers();
+				int32 NumCharactersFound = 0;
+				int32 NumCharactersExpected = GetNumberOfServerWorkers();
 				UWorld* World = GetWorld();
-				for (TActorIterator<ACharacter> It(World); It; ++It)
+				for (ACharacter* Character : TActorRange<ACharacter>(World))
 				{
 					++NumCharactersFound;
 				}
 
-				if (NumCharactersFound == NumCharactersExpected)
-				{
-					FinishStep();
-				}
+				RequireEqual_Int(NumCharactersFound, NumCharactersExpected,
+								 TEXT("Clients should observe the correct number of characters in their world."));
+				FinishStep();
 			},
 			5.0f);
 	}
 
 	{ // Step 3 - Destroy by all servers that have authority
 		AddStep(
-			TEXT("SERVER_ALL_RegisterAutoDestroyActors"), FWorkerDefinition::AllServers,
-			[this]() -> bool {
-				int NumCharactersFound = 0;
-				int NumCharactersExpected = 1;
+			TEXT("SERVER_ALL_RegisterAutoDestroyActors"), FWorkerDefinition::AllServers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				int32 NumCharactersFound = 0;
+				int32 NumCharactersExpected = 1; // 1, because we filter by authority
+				ACharacter* FoundAuthoritativeCharacter = nullptr;
 				UWorld* World = GetWorld();
-				for (TActorIterator<ACharacter> It(World, ACharacter::StaticClass()); It; ++It)
+				for (ACharacter* Character : TActorRange<ACharacter>(World))
 				{
-					if (It->HasAuthority())
+					if (Character->HasAuthority())
 					{
 						++NumCharactersFound;
+						FoundAuthoritativeCharacter = Character;
 					}
 				}
-
-				return NumCharactersFound == NumCharactersExpected;
-			},
-			[this]() {
-				UWorld* World = GetWorld();
-				for (TActorIterator<ACharacter> It(World); It; ++It)
+				if (RequireEqual_Int(NumCharactersFound, NumCharactersExpected,
```

Comment:
Very minor: If you kept a vector of `FoundAuthoritativeCharacter `and check its length instead I think it communicates its intent a bit better. You now have strong implicit link between `NumCharactersExpected `and how you store `FoundAuthoritativeCharacter`.